src/lib/secrets/wallets.ml

```diff
@@ -0,0 +1,96 @@
+open Core
+open Async
+module Secret_keypair = Keypair
+open Signature_lib
+
+(* A simple cache on top of the fs *)
+type t = {mutable cache: Keypair.t list; path: string}
+
+(* TODO: Don't just generate bad passwords *)
+let password = lazy (Deferred.Or_error.return (Bytes.of_string ""))
+
+let load ~logger ~disk_location : t Deferred.t =
+  let logger =
+    Logger.extend logger [("wallets_context", `String "Wallets.get")]
+  in
+  let path = disk_location ^/ "store" in
+  let%bind () = File_system.create_dir path in
+  let%bind handle = Unix.opendir path in
+  let rec go () =
+    match%bind Unix.readdir_opt handle with
```

Comment:
This isn't async so it will block other parts of Coda while it loads. I wish there were a readdir that returned a `Deferred.t` but that doesn't seem to exist. Please correct me if I'm wrong.

Reply:
Really, I am pretty sure `Async.Sys.readdir` returns an Deferred.

This is what utop gives me:

```
utop # Sys.readdir;;
- : string -> string array Deferred.t = <fun>
```